models/pooltogether/ethereum/pooltogether_v4_ethereum_prize_structure.sql

```diff
@@ -0,0 +1,105 @@
+{{ config(
+  alias ='prize_structure',
+  post_hook='{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "pooltogether_v4",
+                                \'["bronder"]\') }}'
+)}}
+
+   WITH
+  --Calculate proze structure for Ethereum network per drawID
+  prizeDistribution AS (
+    --ETHEREUM POST DPR
+    SELECT
+      call_tx_hash AS tx_hash,
+      call_block_time AS block_time,
+      'Ethereum' AS network,
+      drawId,
+      get_json_object(output_0, '$.bitRangeSize') AS bitRange,
+      substr(
+        get_json_object(output_0, '$.tiers'),
+        2,
+        length(get_json_object(output_0, '$.tiers')) -2
+      ) AS tiers,
+      CAST(get_json_object(output_0, '$.dpr') AS int) / power(10, 6) AS dpr,
+      CAST(get_json_object(output_0, '$.prize') AS double) / power(10, 6) AS prize
+    FROM
+      {{ source('pooltogether_v4_ethereum', 'PrizeTierHistoryV2_call_getPrizeTier')}}
+    WHERE
+      call_success = true
+      --ETHEREUM PRE DPR
+    UNION ALL
+    SELECT
+      evt_tx_hash AS tx_hash,
+      evt_block_time,
+      'Ethereum' AS network,
+      drawId,
+      get_json_object(prizeDistribution, '$.bitRangeSize') AS bitRange,
```

Comment:
Maybe this view can be modified to an incremental view.
@BRONDERBRO could you help me to define the unique_key combination for this view?
I may help you with the incremental modification.

Reply:
Thanks @Hosuke I saw that views could be incremental or not, but didn't understand 100% when one is better thatn the other.

In this case, the tx_hash and network fields uniquley identify the records. So those 2 fields should be enough.

Reply:
Thanks, noted for the future!